kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -287,13 +287,21 @@ object KyuubiConf {
       " <li>NOSASL: raw transport.</li>" +
       " <li>NONE: no authentication check.</li>" +
       " <li>KERBEROS: Kerberos/GSSAPI authentication.</li>" +
+      " <li>CUSTOM: User defined authentication.</li>" +
       " <li>LDAP: Lightweight Directory Access Protocol authentication.</li></ul>")
     .version("1.0.0")
     .stringConf
     .transform(_.toUpperCase(Locale.ROOT))
     .checkValues(AuthTypes.values.map(_.toString))
     .createWithDefault(AuthTypes.NONE.toString)
 
+  val AUTHENTICATION_CUSTOM_CLASS: OptionalConfigEntry[String] =
+    buildConf("authentication.custom.class")
+    .doc("User defined authentication class.")
```

Comment:
User-defined authentication implementation of   org.apache.kyuubi.service.authentication.PasswdAuthenticationProvider